core-tests/src/test/java/org/mule/runtime/core/processor/chain/DefaultMessageProcessorChainTestCase.java

```diff
@@ -783,13 +709,11 @@ public void testAll() throws Exception {
   @Test
   public void testChoice() throws Exception {
     ChoiceRouter choiceRouter = new ChoiceRouter();
-    choiceRouter.addRoute(getAppendingMP("1"), new AcceptAllFilter());
-    choiceRouter.addRoute(getAppendingMP("2"), new AcceptAllFilter());
-    choiceRouter.addRoute(getAppendingMP("3"), new AcceptAllFilter());
+    choiceRouter.addRoute(newChain(getAppendingMP("1")), new AcceptAllFilter());
```

Comment:
This is needed as ChoiceRouter does not currently ensure all routes are chains. I didn't update the router, as there is an open question about who should be constructing these chains.